### ISSUE TITLE
[question] `impl Trait` in return position is experimental (see issue #34511)

### ISSUE THREAD

**Issue by gaocegege** [dyweb/dy-weekly-generator#5]
```
error[E0658]: `impl Trait` in return position is experimental (see issue #34511)
  --> src/github.rs:17:13
   |
17 | ) -> Result<impl Iterator<Item = String>, Error> {
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to previous error
```

I tried to build the binary using `cargo run` while I got such a error. Does it a feature, or a bug?

**Comment by gaocegege** [comment]
Ref https://github.com/rust-lang/rust/issues/34511

**Comment by codeworm96** [comment]
What version of Rust are you using (`rustc --version`)?

**Comment by gaocegege** [comment]
rustc 1.25.0 (84203cac6 2018-03-25)

**Comment by gaocegege** [comment]
I am trying to reinstall nightly build...

**Comment by gaocegege** [comment]
rustc 1.27.0-nightly (8a37c75a3 2018-05-02) works for me now...

**Comment by codeworm96** [comment]
Being able to be compiled with stable rust is one of the aims of the project.

This feature has been stabilized in the nightly channel and hope to be released in the next stable version. Just leave the issue open while waiting for the next stable version.

**Comment by gaocegege** [comment]
Why do we reopen it? Are you going to fix it? :惊讶:

**Comment by gaocegege** [comment]
OK, got it...

**Comment by codeworm96** [comment]
The fix is easy, but let's wait for the 1.26 planned at next week.

**Comment by codeworm96** [comment]
This part of code is refactored in #11
